<a href="https://colab.research.google.com/github/Makarena72/1/blob/main/%D0%97%D0%B0%D0%BD%D1%8F%D1%82%D0%B8%D1%8F%20%D0%B2%20%D0%B8%D0%BD%D0%BD%D0%BE%D0%BF%D0%BE%D0%BB%D0%B8%D1%81%D0%B5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
data = pd.read_csv("Motor_Vehicle_Collisions_-_Crashes.csv")
data

CRASH DATE CRASH TIME   BOROUGH  ZIP CODE   LATITUDE  LONGITUDE  \
0      09/11/2021       2:39       NaN       NaN        NaN        NaN   
1      03/26/2022      11:45       NaN       NaN        NaN        NaN   
2      06/29/2022       6:55       NaN       NaN        NaN        NaN   
3      09/11/2021       9:35  BROOKLYN   11208.0  40.667202 -73.866500   
4      12/14/2021       8:13  BROOKLYN   11233.0  40.683304 -73.917274   
...           ...        ...       ...       ...        ...        ...   
44387  02/20/2022       7:35    QUEENS   11412.0  40.698566 -73.769585   
44388  02/20/2022       8:00  BROOKLYN   11226.0  40.641468 -73.959380   
44389  01/08/2022       9:08  BROOKLYN   11228.0  40.617813 -74.006440   
44390  01/08/2022      19:45  BROOKLYN   11223.0  40.607920 -73.970695   
44391  01/08/2022      15:56       NaN       NaN  40.804417 -73.955376   

                      LOCATION           ON STREET NAME CROSS STREET NAME  \
0                          NaN    WHITESTONE EXPRESSWAY         20 AVENUE   
1                          NaN  QUEENSBORO BRIDGE UPPER               NaN   
2                          NaN       THROGS NECK BRIDGE               NaN   
3        (40.667202, -73.8665)                      NaN               NaN   
4      (40.683304, -73.917274)          SARATOGA AVENUE    DECATUR STREET   
...                        ...                      ...               ...   
44387  (40.698566, -73.769585)             ILION AVENUE       WOOD STREET   
44388   (40.641468, -73.95938)          DORCHESTER ROAD      OCEAN AVENUE   
44389   (40.617813, -74.00644)                14 AVENUE         77 STREET   
44390   (40.60792, -73.970695)                      NaN               NaN   
44391  (40.804417, -73.955376)          WEST 116 STREET               NaN   

               OFF STREET NAME  ...   CONTRIBUTING FACTOR VEHICLE 2  \
0                          NaN  ...                     Unspecified   
1                          NaN  ...                             NaN   
2                          NaN  ...                     Unspecified   
3      1211      LORING AVENUE  ...                             NaN   
4                          NaN  ...                             NaN   
...                        ...  ...                             ...   
44387                      NaN  ...                     Unspecified   
44388                      NaN  ...                     Unspecified   
44389                      NaN  ...  Driver Inattention/Distraction   
44390  1740      EAST 3 STREET  ...                             NaN   
44391                      NaN  ...                     Unspecified   

       CONTRIBUTING FACTOR VEHICLE 3  CONTRIBUTING FACTOR VEHICLE 4  \
0                                NaN                            NaN   
1                                NaN                            NaN   
2                                NaN                            NaN   
3                                NaN                            NaN   
4                                NaN                            NaN   
...                              ...                            ...   
44387                            NaN                            NaN   
44388                            NaN                            NaN   
44389                    Unspecified                            NaN   
44390                            NaN                            NaN   
44391                            NaN                            NaN   

       CONTRIBUTING FACTOR VEHICLE 5  COLLISION_ID  \
0                                NaN       4455765   
1                                NaN       4513547   
2                                NaN       4541903   
3                                NaN       4456314   
4                                NaN       4486609   
...                              ...           ...   
44387                            NaN       4504317   
44388                            NaN       4504085   
4

In [3]:
data.info()
data.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 44392 entries, 0 to 44391
Data columns (total 29 columns):
 #   Column                         Non-Null Count  Dtype  
---  ------                         --------------  -----  
 0   CRASH DATE                     44392 non-null  object 
 1   CRASH TIME                     44392 non-null  object 
 2   BOROUGH                        29083 non-null  object 
 3   ZIP CODE                       29077 non-null  float64
 4   LATITUDE                       40934 non-null  float64
 5   LONGITUDE                      40934 non-null  float64
 6   LOCATION                       40934 non-null  object 
 7   ON STREET NAME                 32326 non-null  object 
 8   CROSS STREET NAME              20499 non-null  object 
 9   OFF STREET NAME                12066 non-null  object 
 10  NUMBER OF PERSONS INJURED      44392 non-null  int64  
 11  NUMBER OF PERSONS KILLED       44392 non-null  int64  
 12  NUMBER OF PEDESTRIANS INJURED  44392 non-null 

ZIP CODE      LATITUDE     LONGITUDE  NUMBER OF PERSONS INJURED  \
count  29077.000000  40934.000000  40934.000000               44392.000000   
mean   10893.654469     40.468390    -73.452172                   0.474072   
std      522.718396      3.217758      5.839102                   0.799830   
min    10000.000000      0.000000    -74.251840                   0.000000   
25%    10457.000000     40.665510    -73.962265                   0.000000   
50%    11208.000000     40.712864    -73.919980                   0.000000   
75%    11238.000000     40.783964    -73.867489                   1.000000   
max    11697.000000     40.912827      0.000000                  18.000000   

       NUMBER OF PERSONS KILLED  NUMBER OF PEDESTRIANS INJURED  \
count              44392.000000                   44392.000000   
mean                   0.002838                       0.070508   
std                    0.055684                       0.268964   
min                    0.000000                       0.000000   
25%                    0.000000                       0.000000   
50%                    0.000000                       0.000000   
75%                    0.000000                       0.000000   
max                    3.000000                       6.000000   

       NUMBER OF PEDESTRIANS KILLED  NUMBER OF CYCLIST INJURED  \
count                  44392.000000               44392.000000   
mean                       0.001216                   0.046743   
std                        0.034857                   0.214267   
min                        0.000000                   0.000000   
25%                        0.000000                   0.000000   
50%                        0.000000                   0.000000   
75%                        0.000000                   0.000000   
max                        1.000000                   3.000000   

       NUMBER OF CYCLIST KILLED  NUMBER OF MOTORIST INJURED  \
count              44392.000000                44392.000000   
mean                   0.000180                    0.341796   
std                    0.013423                    0.772680   
min                    0.000000                    0.000000   
25%                    0.000000                    0.000000   
50%                    0.000000                    0.000000   
75%                    0.000000                    0.000000   
max                    1.000000                   18.000000   

       NUMBER OF MOTORIST KILLED  COLLISION_ID  
count               44392.000000  4.439200e+04  
mean                    0.001307  4.450903e+06  
std                     0.039117  3.777946e+04  
min                     0.000000  3.456194e+06  
25%                     0.000000  4.418669e+06  
50%                     0.000000  4.429984e+06  
75%                     0.000000  4.486814e+06  
max                     3.000000  4.573583e+06

В данной базе приводится сводка данных о произошедших ДТП с датами, временем, местонахождением, количеством пострадавших. С помощью команды .info() определено, что в исследуемой базе данных всего 44392 строки и 29 столбцов, 4 из которых имеют формат с плавающей точкой, 7 целочисленных и 17 имеют строковый формат. Также имеется большое количество пустых ячеек, особенно в столбцах CONTRIBUTING FACTOR VEHICLE 5 и VEHICLE TYPE CODE 5.  Максимальное количество пострадавших в аварии - 18 человек, из которых все являются водителями.

In [ ]:
1+1